base/terminfo_data.jl

```diff
@@ -17,524 +32,744 @@ Specification of a single terminal capability.
+    TermCapability(:pc_term_options,           :pctrm,    "PC terminal options"),
+    TermCapability(:scancode_escape,           :scesc,    "Escape for scancode emulation"),
+    TermCapability(:alt_scancode_esc,          :scesa,    "Alternate escape for scancode emulation"),
+    TermCapability(:enter_horizontal_hl_mode,  :ehhlm,    "Enter horizontal highlight mode"),
+    TermCapability(:enter_left_hl_mode,        :elhlm,    "Enter left highlight mode"),
+    TermCapability(:enter_low_hl_mode,         :elohlm,   "Enter low highlight mode"),
+    TermCapability(:enter_right_hl_mode,       :erhlm,    "Enter right highlight mode"),
+    TermCapability(:enter_top_hl_mode,         :ethlm,    "Enter top highlight mode"),
+    TermCapability(:enter_vertical_hl_mode,    :evhlm,    "Enter vertical highlight mode"),
+    TermCapability(:set_a_attributes,          :sgr1,     "Define second set of video attributes #1-#6"),
+    TermCapability(:set_pglen_inch,            :slength,  "Set page length to #1 hundredth of an inch (some implementations use sL for termcap)."),
+    TermCapability(:termcap_init2,             :OTi2,     "secondary initialization string"),
+    TermCapability(:termcap_reset,             :OTrs,     "terminal reset string"),
+    TermCapability(:linefeed_if_not_lf,        :OTnl,     "use to move down"),
+    TermCapability(:backspace_if_not_bs,       :OTbc,     "move left, if not ^H"),
+    TermCapability(:other_non_function_keys,   :OTko,     "list of self-mapped keycaps"),
+    TermCapability(:arrow_key_map,             :OTma,     "map motion-keys for vi version 2"),
+    TermCapability(:acs_ulcorner,              :OTG2,     "single upper left"),
+    TermCapability(:acs_llcorner,              :OTG3,     "single lower left"),
+    TermCapability(:acs_urcorner,              :OTG1,     "single upper right"),
+    TermCapability(:acs_lrcorner,              :OTG4,     "single lower right"),
+    TermCapability(:acs_ltee,                  :OTGR,     "tee pointing right"),
+    TermCapability(:acs_rtee,                  :OTGL,     "tee pointing left"),
+    TermCapability(:acs_btee,                  :OTGU,     "tee pointing up"),
+    TermCapability(:acs_ttee,                  :OTGD,     "tee pointing down"),
+    TermCapability(:acs_hline,                 :OTGH,     "single horizontal line"),
+    TermCapability(:acs_vline,                 :OTGV,     "single vertical line"),
+    TermCapability(:acs_plus,                  :OTGC,     "single intersection"),
+    TermCapability(:memory_lock,               :meml,     "lock memory above cursor"),
+    TermCapability(:memory_unlock,             :memu,     "unlock memory"),
+    TermCapability(:box_chars_1,               :box1,     "box characters primary set"),
 ]
+
+"""
+Terminfo extensions that NCurses 6.4 is aware of.
+"""
+const TERM_USER = Dict{Tuple{Type, Symbol}, Tuple{Union{Nothing, Symbol}, String}}(
```

Comment:
It is typically preferable to keep things in terms of concrete types as much as possible
```suggestion
const TERM_USER = Dict{Tuple{DataType, Symbol}, Union{Tuple{Nothing, String}, Tuple{Symbol, String}}}(
```

Reply:
Sometimes doing instead a special token value (`:nothing`) rather than using nullable logic might also be simpler? But doesn't seem that necessarily applies here, so no change is actually required (e.g. to make them all `:nothing` instead of `nothing`.)

Reply:
Made more concrete.